src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -237,20 +237,68 @@ public StudentParticipation startExercise(Exercise exercise, User user) {
+     * @param user the user for which to create a participation and submission
+     * @param submissionType the type of submission to create if none exist yet
+     * @return the participation connecting the given exercise and user
+     */
+    public StudentParticipation createParticipationWithEmptySubmissionIfNotExisting(Exercise exercise, User user, SubmissionType submissionType) {
+        Optional<StudentParticipation> optionalStudentParticipation = findOneByExerciseIdAndStudentLoginAnyState(exercise.getId(), user.getLogin());
+        StudentParticipation participation;
+        if (optionalStudentParticipation.isEmpty()) {
+            // create a new participation only if no participation can be found
+            if (exercise instanceof ProgrammingExercise) {
+                participation = new ProgrammingExerciseStudentParticipation();
+            }
+            else {
+                participation = new StudentParticipation();
+            }
+            participation.setInitializationState(FINISHED);
+            participation.setInitializationDate(ZonedDateTime.now());
+            participation.setExercise(exercise);
+            participation.setStudent(user);
+        }
+        else {
+            // make sure participation and exercise are connected
+            participation = optionalStudentParticipation.get();
+            participation.setExercise(exercise);
+        }
+        participation = save(participation);
+
+        // initialize a programming, modeling, text or file upload submission (depending on the exercise type), it will not do anything in the case of a quiz exercise
+        if (optionalStudentParticipation.isEmpty() || !submissionRepository.existsByParticipationId(participation.getId())) {
+            participation = (StudentParticipation) findOneWithEagerSubmissions(participation.getId());
+            initializeSubmission(participation, exercise, true, submissionType);
+            participation = save(participation);
+        }
+
+        return participation;
+    }
+
     /**
      * Initializes a new text, modeling or file upload submission (depending on the type of the given exercise), connects it with the given participation and stores it in the
      * database.
      *
-     * @param participation the participation for which the submission should be initialized
-     * @param exercise      the corresponding exercise, should be either a text, modeling or file upload exercise, otherwise it will instantly return and not do anything
+     * @param participation                 the participation for which the submission should be initialized
+     * @param exercise                      the corresponding exercise, should be either a text, modeling or file upload exercise, otherwise it will instantly return and not do anything
+     * @param allowProgrammingExercise      whether to explicitly allow the submission initialization for programming exercises
```

Comment:
As far as I could see it, we never have the case that this is false and the exercise is a programming exercise. The method you created in line 324 only gets called if the exercise is not a programing exercise. Besides, because you created an overloaded method here, you hide the fact that the alternative in line 324 does not initialize programming submissions. I would suggest to just remove the overloaded variant. If a developer does not want to initialize a programming submission in this case, he should not call it at all.

Reply:
Fair enough. I was a bit hesitant to just change the current behavior of `initializeSubmission` (since it explicitly does nothing for programming exercises at the moment) and thus went with the overloaded variant but you are right that there are no such usages at the moment, so I'll refactor it as you described. The responsibility will then be on the user of this method, if he passes in a programming exercise, he should indeed not be surprised if it actually initializes a submission for it.